.github/workflows/publish.yml

```diff
@@ -3,6 +3,8 @@ name: Upload Package to Pypi and TestPyPI
 on:
   push:
     branches: [ main ]
+    tags:
+      - '*'
```

Comment:
This looks really clean.

I wonder if we should have 2 actions then - one for 'push: branches' that publishes to the test pipy, and one with 'push: tags' here that publishes to the real one. That way we can get rid to the 'if ref/tags' logic below, since it doesn't seem like it worked anyway?

Reply:
After discussing on IRC, the clean-up will be handled in a follow-up PR.